runtime-sdk/modules/evm/src/types.rs

```diff
@@ -0,0 +1,168 @@
+    }
+}
+
+impl AsRef<[u8]> for H256 {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+
+impl cbor::Encode for H256 {
+    fn into_cbor_value(self) -> cbor::Value {
+        cbor::Value::ByteString(self.0.as_bytes().to_vec())
+    }
+}
+
+impl cbor::Decode for H256 {
+    fn try_from_cbor_value(value: cbor::Value) -> Result<Self, cbor::DecodeError> {
+        match value {
+            cbor::Value::ByteString(data) => Ok(Self::from_slice(&data)),
+            _ => Err(cbor::DecodeError::UnexpectedType),
+        }
+    }
+}
+
+impl H256 {
+    pub fn from_slice(src: &[u8]) -> Self {
+        H256(primitive_types::H256::from_slice(src))
+    }
+
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+pub struct U256(primitive_types::U256);
+
+impl From<primitive_types::U256> for U256 {
+    fn from(a: primitive_types::U256) -> U256 {
+        U256(a)
+    }
+}
+
+impl From<U256> for primitive_types::U256 {
+    fn from(a: U256) -> primitive_types::U256 {
+        a.0
+    }
+}
+
+impl cbor::Encode for U256 {
+    fn into_cbor_value(self) -> cbor::Value {
+        let mut bs = [0u8; 32];
+        self.0.to_big_endian(&mut bs);
+        cbor::Value::ByteString(bs.to_vec())
+    }
+}
+
+impl cbor::Decode for U256 {
+    fn try_from_cbor_value(value: cbor::Value) -> Result<Self, cbor::DecodeError> {
+        match value {
+            cbor::Value::ByteString(data) => {
+                if data.len() != 32 {
+                    Err(cbor::DecodeError::UnexpectedType)
+                } else {
+                    Ok(Self::from_big_endian(&data))
+                }
+            }
+            _ => Err(cbor::DecodeError::UnexpectedType),
+        }
+    }
+}
+
+impl U256 {
+    pub fn from_big_endian(src: &[u8]) -> Self {
+        U256(primitive_types::U256::from_big_endian(src))
+    }
+}
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+pub struct H160(primitive_types::H160);
+
+impl From<primitive_types::H160> for H160 {
+    fn from(a: primitive_types::H160) -> H160 {
+        H160(a)
+    }
+}
+
+impl From<H160> for primitive_types::H160 {
+    fn from(a: H160) -> primitive_types::H160 {
+        a.0
+    }
+}
+
+impl AsRef<[u8]> for H160 {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+
+impl cbor::Encode for H160 {
+    fn into_cbor_value(self) -> cbor::Value {
+        cbor::Value::ByteString(self.0.as_bytes().to_vec())
+    }
+}
+
+impl cbor::Decode for H160 {
+    fn try_from_cbor_value(value: cbor::Value) -> Result<Self, cbor::DecodeError> {
+        match value {
+            cbor::Value::ByteString(data) => Ok(Self::from_slice(&data)),
+            _ => Err(cbor::DecodeError::UnexpectedType),
+        }
+    }
+}
+
+impl H160 {
+    pub fn from_slice(src: &[u8]) -> Self {
+        H160(primitive_types::H160::from_slice(src))
+    }
+
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
```

Comment:
this is a tonne of code that you don't need. Try this instead:

```rust
// This `mod` exists solely to place an `#[allow(warnings)]` around the generated code.
#[allow(warnings)]
mod eth {
    use super::*;

    macro_rules! construct_fixed_hash {
        ($name:ident($num_bytes:literal)) => {
            fixed_hash::construct_fixed_hash! {
                #[derive(cbor::Encode, cbor::Decode)
               #[cbor(transparent)
                pub struct $name($num_bytes);
            }
        };
    }

    macro_rules! construct_uint {
        ($name:ident($num_words:tt)) => {
            uint::construct_uint! {
                #[derive(cbor::Encode, cbor::Decode)]
                #[cbor(transparent)]
                pub struct $name($num_words);
            }
        };
    }

    construct_fixed_hash!(H160(20));
    construct_fixed_hash!(H256(32));
    construct_uint!(U256(4));

    macro_rules! impl_upstream_conversions {
        ($($ty:ident => $upstream:ident),* $(,)?) => {
            $(
                impl From<$ty> for primitive_types::$upstream {
                    fn from(t: $ty) -> Self {
                        Self(t.0)
                    }
                }

                impl From<primitive_types::$upstream> for $ty {
                    fn from(t: primitive_types::$upstream) -> Self {
                        Self(t.0)
                    }
                }
            )*
        }
    }

    impl_upstream_conversions! {
        H160, H256, U256
    }
}
pub use eth::{H160, H256, U256};
```
and then you can remove all of the converters, and also use strongly-typed values in the RPC structs

Reply:
Nice! Thanks!